CONTRIBUTING.rst

```diff
@@ -75,15 +80,43 @@ Check the commits you have on your branch and then push them to your forked repo
-If you're proposing a **patch** that needs to be backported to a specific branch, please make sure to mention it in your pull request, such that the project responsible can properly label it. If your pull request requires further effort, please use the labels "Do not merge yet" and "Work in progress".
-The pull request title has to be meaningful as well, even though it's not used for the release notes. You might want to describe your changes and the reason behind that, it's quite helpful when we need to check a module's history.
+**Step 8**: then create a pull request either from your fork, or from the official github repository. There is a pull request template that you need to edit/update before actually creating your pull request:
+
+- please make sure to provide a meaningful and short PR title;
+- also provide a short description of the changes provided in the PR (if useful, also provide a reason for that decision and implementation)
+- provide any possible dependency and/or external changes that are required to go with your pull request (kubernetes, COMP packages, deployments, etc).
+If your pull request requires further effort and/or it's still a work in progress, please use one or both labels like: "Do not merge yet" and "Work in progress". If you're proposing a **patch** that needs to be backported to a specific branch and/or pushed to production right away, then please make sure to mention it in your pull request such that the proper actions and labelling are done.
+
+**Step 9**: before asking for a pull request review, please make sure to provide any other material that is necessary to go together with your WMCore contribution (deployment, configurations, kubernetes, etc).
+Once your pull request is ready to be reviewed, use the `Reviewers` option to ask a specific person(s) to review it. Watch your pull request for comments and feedback.
+If further changes are required in your pull request, you might want to provide them in a separate commit, making it easier to review only the latest differences.
+
+**Step 10**: when your pull request gets approved by at least one reviewer, you must squash your commits according to what has been explained above. In short:
 
-**Step 9**: once your pull request is ready to be reviewed, use the `Reviewers` option to ask a specific person(s) to review it. Watch your pull request for comments and feedback.
-If further changes are required to your pull request, you might want to provide them in a separate commit, making it easier to review only the latest difference.
+- logical and algorithmical changes go in the first commit;
+- while unit tests, pylint reformat, pep conventions, aesthetic minor improves go in the second commit.
+Make sure to provide meaningful - and short - commit messages, since they are used in our software release notes
```

Comment:
We'd better add here a small comment about commit messages amendment upon commit squashing (if needed), because, while the squashed commit does keep all the commit messages from the whole PR, it is only the top one that stays visible, and in many cases this is not the most representative one for the changes provided with the PR

Reply:
This is implicit, but I agree that we can make this an explicit comment as well. Updating.